Assets/MixedRealityToolkit-SDK/Features/Input/FocusProvider.cs

```diff
@@ -220,34 +198,81 @@ public override int GetHashCode()
             }
         }
 
-        #region MonoBehaviour Implementation
+        #region IMixedRealityService Implementation
 
-        private async void Awake()
+        /// <inheritdoc />
+        public override void Initialize()
         {
             if (uiRaycastCamera == null)
             {
-                Debug.LogWarning("No UIRaycastCamera assigned! Creating a new UIRaycastCamera.\n" +
-                                 "To create a UIRaycastCamera in your scene, find this Focus Provider GameObject and add one there.");
-                CreateUiRaycastCamera();
+                EnsureUiRaycastCameraSetup();
             }
 
-            await WaitUntilInputSystemValid;
+            if (MixedRealityToolkit.InputSystem == null)
```

Comment:
Can we break these calls in to a generic function, I hate duplicated code :D

Reply:
If you can figure out a way to do it, sure. I'm accepting PRs.